MerlinAU.sh

```diff
@@ -2561,6 +2619,348 @@ _GetLatestFWUpdateVersionFromWebsite_()
     return 0
 }
 
+##---------------------------------------##
+## Added by ExtremeFiretop [2024-Feb-23] ##
+##---------------------------------------##
+_GetLatestFWUpdateVersionFromGithub_()
+{
+    local url="$1"  # GitHub API URL for the latest release
+    local firmware_type="$2"  # Type of firmware, e.g., "tuf", "rog" or "pure"
+
+    local search_type="$firmware_type"  # Default to the input firmware_type
+
+    # If firmware_type is "pure", set search_type to include "squashfs" as well
+    if [ "$firmware_type" = "pure" ]; then
+        search_type="pure\|squashfs\|ubi"
+    fi
+
+    # Fetch the latest release data from GitHub
+    local release_data=$(curl -s "$url")
+
+    # Construct the grep pattern based on search_type
+    local grep_pattern="\"browser_download_url\": \".*${PRODUCT_ID}.*\(${search_type}\).*\.\(w\|pkgtb\)\""
+
+    # Filter the JSON for the desired firmware using grep and head to fetch the URL
+    local download_url=$(echo "$release_data" | 
+        grep -o "$grep_pattern" | 
+        grep -o "https://[^ ]*\.\(w\|pkgtb\)" | 
+        head -1)
+
+    # Check if a URL was found
+    if [ -z "$download_url" ]; then
+        echo "**ERROR** **NO_GITHUB_URL**"
+        return 1
+    else
+        # Extract the version from the download URL or release data
+        local version=$(echo "$download_url" | grep -oE "$PRODUCT_ID[_-][0-9.]+[^/]*" | sed "s/${PRODUCT_ID}[_-]//;s/.w$//;s/_/./g")
```

Comment:
Line **2655**: use enclosing curly braces to clearly separate the "`$PRODUCT_ID`" variable reference in the **`grep`** command pattern syntax:

```
| grep -oE "${PRODUCT_ID}[_-][0-9.]+[^/]*" |
```

Reply:
Reviewing machineee

Reply:
Done : https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/pull/186/commits/f8a345d913fd8648cdd59fe4c1ab69f16f086eb6